test/t8_geometry/t8_gtest_geometry.cxx

```diff
@@ -0,0 +1,296 @@
+{
+  /* TODO: Add a test for the jacobian, as soon as its implemented. */
+
+  t8_debugf ("Testing linear geometry evaluation.\n");
+
+  /* Create random points in [0,1]^d and check if they are mapped correctly. */
+  t8_geometry_linear  linear_geom (dim);
+  t8_cmesh_t          cmesh;
+
+  int                 num_points = 10000;
+  double              point[3];
+  double              point_mapped[3];
+  int                 seed = 0; /* RNG seed */
+  t8_eclass_t         tree_class;
+  const t8_geometry_c *cmesh_geom;
+
+  /* Build a one tree cmesh on the unit square with linear geometry. */
+  switch (dim) {
+  case 0:
+    tree_class = T8_ECLASS_VERTEX;
+    break;
+  case 1:
+    tree_class = T8_ECLASS_LINE;
+    break;
+  case 2:
+    tree_class = T8_ECLASS_QUAD;
+    break;
+  case 3:
+    tree_class = T8_ECLASS_HEX;
+    break;
+  default:
+    SC_ABORT_NOT_REACHED ();
+  }
+  cmesh = t8_cmesh_new_hypercube (tree_class, sc_MPI_COMM_WORLD, 0, 0, 0);
+
+  /* Double check that the geometry is the linear geometry. */
+  cmesh_geom = t8_cmesh_get_tree_geometry (cmesh, 0);
+  ASSERT_TRUE (cmesh_geom != NULL) << "Could not get cmesh's geometry.";
+  int                 has_same_name = strcmp (cmesh_geom->t8_geom_get_name (),
+                                              linear_geom.t8_geom_get_name
+                                              ());
+  ASSERT_FALSE (has_same_name) <<
+    "cmesh's geometry is not the linear geometry.";
+
+  srand (seed);
+  for (int ipoint = 0; ipoint < num_points; ++ipoint) {
+    /* Compute random coordinates in [0,1].
+     * These are seen as reference coordinates in the single
+     * cmesh tree. Our geometry will map them into the physical
+     * space. Since this space is also [0,1] and the cmesh only
+     * has one tree, the mapped coordinates must be the same as the 
+     * reference coordinates. */
+    point[0] = (double) rand () / RAND_MAX;
+    point[1] = (double) rand () / RAND_MAX;
+    point[2] = (double) rand () / RAND_MAX;
+
+    /* Evaluate the geometry */
+    t8_geometry_evaluate (cmesh, 0, point, point_mapped);
+    /* Check that the first dim coordinates are the same */
+    int                 idim;
+    for (idim = 0; idim < dim; ++idim) {
+      const double        tolerance = 1e-14;
+      ASSERT_LT (fabs (point[idim] - point_mapped[idim]),
+                 tolerance) << "Linear geometry computed wrong value.";
+    }
+    /* Check that the remaining entries are 0. */
+    for (; idim < 3; ++idim) {
+      ASSERT_EQ (point_mapped[idim],
+                 0) << "Linear geometry computed wrong value.";
+    }
+  }
+  /* Destroy the cmesh */
+  t8_cmesh_destroy (&cmesh);
+}
+
+/* *INDENT-OFF* */
+INSTANTIATE_TEST_SUITE_P (t8_gtest_geometry, geometry,testing::Range(0,4));
+/* *INDENT-ON* */
+
+TEST (test_geometry, cmesh_geometry)
+{
+  t8_cmesh_t          cmesh;
+  /* TODO: We need to use a different geometry since we remove the identity geomeytry. */
```

Comment:
I have no idea what this comment is about. Trying to figure this out and if we need to change something here.

Reply:
i wrote this tesr initially for the identity geom. a geom that is now removed.
since tha test now uses linear and zero geom, we can remove this comment.

Some comments below need to be adapted though since they still talk abuot identity geom.